Index: lumosapi_client/README
===================================================================
--- /dev/null
+++ lumosapi_client/README
@@ -0,0 +1 @@
+This is the LLNL patched version of the LumOS client
Index: lumosapi_client/lumosapi_client/rest.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/rest.py
+++ lumosapi_client/lumosapi_client/rest.py
@@ -18,6 +18,7 @@ import re
 import ssl
 
 import urllib3
+import requests
 
 from lumosapi_client.exceptions import ApiException, ApiValueError
 
@@ -70,6 +71,12 @@ class RESTClientObject:
         # https://github.com/shazow/urllib3/blob/f9409436f83aeb79fbaf090181cd81b784f1b8ce/urllib3/connectionpool.py#L680  # noqa: E501
         # Custom SSL certificates and client certificates: http://urllib3.readthedocs.io/en/latest/advanced-usage.html  # noqa: E501
 
+        context = ssl._create_unverified_context()
+        context.minimum_version = ssl.TLSVersion.TLSv1_2
+        context.maximum_version = ssl.TLSVersion.TLSv1_3
+        context.options = ssl.PROTOCOL_TLS & ssl.OP_NO_TLSv1 & ssl.OP_NO_TLSv1_1
+        requests.packages.urllib3.disable_warnings(requests.packages.urllib3.exceptions.InsecureRequestWarning)
+
         # cert_reqs
         if configuration.verify_ssl:
             cert_reqs = ssl.CERT_REQUIRED
@@ -81,7 +88,7 @@ class RESTClientObject:
             "ca_certs": configuration.ssl_ca_cert,
             "cert_file": configuration.cert_file,
             "key_file": configuration.key_file,
-            "ca_cert_data": configuration.ca_cert_data,
+            "ssl_context": context,
         }
         if configuration.assert_hostname is not None:
             pool_args['assert_hostname'] = (
Index: lumosapi_client/lumosapi_client/models/fru_list.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/fru_list.py
+++ lumosapi_client/lumosapi_client/models/fru_list.py
@@ -23,6 +23,8 @@ from lumosapi_client.models.fru import F
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class FRUList(BaseModel):
     """
     List of Field Replaceable Units
@@ -47,7 +49,7 @@ class FRUList(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/package.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/package.py
+++ lumosapi_client/lumosapi_client/models/package.py
@@ -24,6 +24,8 @@ from lumosapi_client.models.firmware imp
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class Package(BaseModel):
     """
     Package
@@ -49,7 +51,7 @@ class Package(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/datetime_encoder.py
===================================================================
--- /dev/null
+++ lumosapi_client/lumosapi_client/models/datetime_encoder.py
@@ -0,0 +1,18 @@
+# coding: utf-8
+
+import json
+import datetime
+
+class DateTimeEncoder(json.JSONEncoder):
+    def default(self, z):
+        if isinstance(z, datetime.datetime):
+            try:
+                local_time = z.astimezone()
+            except OverflowError as e:
+                # Cannot convert to local time zone
+                local_time = z
+            return (str(local_time))
+        if isinstance(z, datetime.date):
+            return (str(z))
+        else:
+            return super().default(z)
Index: lumosapi_client/lumosapi_client/models/package_list.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/package_list.py
+++ lumosapi_client/lumosapi_client/models/package_list.py
@@ -23,6 +23,8 @@ from lumosapi_client.models.package impo
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class PackageList(BaseModel):
     """
     A list of packages that have been uploaded to the library, including the currently active package
@@ -47,7 +49,7 @@ class PackageList(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/library_status.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/library_status.py
+++ lumosapi_client/lumosapi_client/models/library_status.py
@@ -26,6 +26,8 @@ from lumosapi_client.models.ntp_status i
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class LibraryStatus(BaseModel):
     """
     General status of library hardware and software. All fields are read-only.
@@ -51,7 +53,7 @@ class LibraryStatus(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/task.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/task.py
+++ lumosapi_client/lumosapi_client/models/task.py
@@ -27,6 +27,8 @@ from lumosapi_client.models.task_types i
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class Task(BaseModel):
     """
     Task Manager related information
@@ -62,7 +64,7 @@ class Task(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/task_list.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/task_list.py
+++ lumosapi_client/lumosapi_client/models/task_list.py
@@ -23,6 +23,8 @@ from lumosapi_client.models.task import
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class TaskList(BaseModel):
     """
     Task Manager related information
@@ -47,7 +49,7 @@ class TaskList(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/status_message.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/status_message.py
+++ lumosapi_client/lumosapi_client/models/status_message.py
@@ -24,6 +24,8 @@ from lumosapi_client.models.severities i
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class StatusMessage(BaseModel):
     """
     Each library status message contains an id and a severity.
@@ -53,7 +55,7 @@ class StatusMessage(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/status_message_list.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/status_message_list.py
+++ lumosapi_client/lumosapi_client/models/status_message_list.py
@@ -23,6 +23,8 @@ from lumosapi_client.models.status_messa
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class StatusMessageList(BaseModel):
     """
     StatusMessageList
@@ -47,7 +49,7 @@ class StatusMessageList(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/library_diagnostic.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/library_diagnostic.py
+++ lumosapi_client/lumosapi_client/models/library_diagnostic.py
@@ -28,6 +28,8 @@ from lumosapi_client.models.task_types i
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class LibraryDiagnostic(BaseModel):
     """
     LibraryDiagnostic includes information about the library diagnostic.
@@ -64,7 +66,7 @@ class LibraryDiagnostic(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/library_diagnostic_list.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/library_diagnostic_list.py
+++ lumosapi_client/lumosapi_client/models/library_diagnostic_list.py
@@ -23,6 +23,8 @@ from lumosapi_client.models.library_diag
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class LibraryDiagnosticList(BaseModel):
     """
     A paginated list of library diagnostics.
@@ -47,7 +49,7 @@ class LibraryDiagnosticList(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/mlm_record.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/mlm_record.py
+++ lumosapi_client/lumosapi_client/models/mlm_record.py
@@ -32,6 +32,8 @@ from lumosapi_client.models.mlm_record_r
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class MLMRecord(BaseModel):
     """
     All Media Lifecycle Management (MLM) data stored on the Medium Auxiliary Memory (MAM) of a tape cartridge
@@ -88,7 +90,7 @@ class MLMRecord(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/mlm_record_read_write.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/mlm_record_read_write.py
+++ lumosapi_client/lumosapi_client/models/mlm_record_read_write.py
@@ -23,6 +23,8 @@ from typing import Any, ClassVar, Dict,
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class MLMRecordReadWrite(BaseModel):
     """
     MLMRecordReadWrite
@@ -56,7 +58,7 @@ class MLMRecordReadWrite(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/mlm_list.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/mlm_list.py
+++ lumosapi_client/lumosapi_client/models/mlm_list.py
@@ -23,6 +23,8 @@ from lumosapi_client.models.mlm_record i
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class MLMList(BaseModel):
     """
     List of Media Lifecycle Management (MLM) data for requested tapes.
@@ -47,7 +49,7 @@ class MLMList(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/humidity_reading.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/humidity_reading.py
+++ lumosapi_client/lumosapi_client/models/humidity_reading.py
@@ -23,6 +23,8 @@ from typing import Any, ClassVar, Dict,
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class HumidityReading(BaseModel):
     """
     HumidityReading
@@ -47,7 +49,7 @@ class HumidityReading(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/library_humidity_metrics.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/library_humidity_metrics.py
+++ lumosapi_client/lumosapi_client/models/library_humidity_metrics.py
@@ -23,6 +23,8 @@ from lumosapi_client.models.humidity_rea
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class LibraryHumidityMetrics(BaseModel):
     """
     LibraryHumidityMetrics
@@ -48,7 +50,7 @@ class LibraryHumidityMetrics(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/library_temperature_metrics.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/library_temperature_metrics.py
+++ lumosapi_client/lumosapi_client/models/library_temperature_metrics.py
@@ -23,6 +23,8 @@ from lumosapi_client.models.temperature_
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class LibraryTemperatureMetrics(BaseModel):
     """
     LibraryTemperatureMetrics
@@ -48,7 +50,7 @@ class LibraryTemperatureMetrics(BaseMode
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
Index: lumosapi_client/lumosapi_client/models/temperature_reading.py
===================================================================
--- lumosapi_client.orig/lumosapi_client/models/temperature_reading.py
+++ lumosapi_client/lumosapi_client/models/temperature_reading.py
@@ -23,6 +23,8 @@ from typing import Any, ClassVar, Dict,
 from typing import Optional, Set
 from typing_extensions import Self
 
+from lumosapi_client.models.datetime_encoder import DateTimeEncoder
+
 class TemperatureReading(BaseModel):
     """
     TemperatureReading
@@ -47,7 +49,7 @@ class TemperatureReading(BaseModel):
     def to_json(self) -> str:
         """Returns the JSON representation of the model using alias"""
         # TODO: pydantic v2: use .model_dump_json(by_alias=True, exclude_unset=True) instead
-        return json.dumps(self.to_dict())
+        return json.dumps(self.to_dict(),cls=DateTimeEncoder)
 
     @classmethod
     def from_json(cls, json_str: str) -> Optional[Self]:
